src/main/java/uk/gov/companieshouse/itemgroupworkflowapi/controller/ItemGroupController.java

```diff
@@ -13,96 +15,118 @@
-    public static final String ITEM_GROUP_ALREADY_EXISTS = "create_item_group: already exists";
+    public static final String CREATE_ITEM_GROUP_ALREADY_EXISTS_PREFIX = "create_item_group: already exists";
     private final LoggingUtils logger;
     private final ItemGroupsService itemGroupsService;
     private final ItemGroupsValidator itemGroupsValidator;
 
-    public ItemGroupController(LoggingUtils logger, ItemGroupsService itemGroupsService, ItemGroupsValidator itemGroupsValidator) {
+    public ItemGroupController(LoggingUtils logger,
+                               ItemGroupsService itemGroupsService,
+                               ItemGroupsValidator itemGroupsValidator) {
         this.logger = logger;
         this.itemGroupsService = itemGroupsService;
         this.itemGroupsValidator = itemGroupsValidator;
     }
-
-    @PostMapping("${uk.gov.companieshouse.itemgroupworkflowapi.createitemgroup}")
-    public ResponseEntity<Object> createItemGroup(final @RequestHeader(REQUEST_ID_HEADER_NAME) String requestId,
+    /**
+     * Create item group from ItemGroupData<p>
+     * POST RequestID logged<p>
+     * Validation performed on item group, on failure return BAD_REQUEST<p>
+     * Item group already exists check performed, if already exists return CONFLICT<p>
+     * Item group created, on success return CREATED, on error return BAD_REQUEST<p>
+     */
+    @PostMapping(path = "${uk.gov.companieshouse.itemgroupworkflowapi.createitemgroup}",
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Object> createItemGroup(final @RequestHeader(X_REQUEST_ID_HEADER_NAME) String xRequestId,
                                                   final @RequestBody ItemGroupData itemGroupData) {
-        logRequestId(requestId);
-        List<String> errors = itemGroupsValidator.validateCreateItemPayload(itemGroupData);
+        logRequestId(xRequestId);
+        List<String> errors = itemGroupsValidator.validateCreateItemData(itemGroupData);
 
         if (!errors.isEmpty()) {
-            return buildValidationResponse(errors, itemGroupData);
+            return buildValidationResponse(xRequestId, errors);
         }
 
         try {
-            if (itemGroupsService.doesItemGroupExist(itemGroupData)) {
-                return buildItemAlreadyExistsResponse(itemGroupData);
-            }
+            if (itemGroupsService.doesItemGroupExist(itemGroupData))
+                return buildItemAlreadyExistsResponse(xRequestId, itemGroupData);
 
             final ItemGroup savedItem = itemGroupsService.createItemGroup(itemGroupData);
-            return buildCreateSuccessResponse(savedItem);
-
-        } catch (Exception ex) {
-            return buildErrorResponse(ex, itemGroupData);
+            return buildCreateSuccessResponse(xRequestId, savedItem);
+        }
+        catch (Exception ex) {
```

Comment:
Are we planning on dealing with exceptions in a finer-grained way in another PR then?

Reply:
Yep, we are ready to add exception handling to deal with internal server errors etc, but potentially under another ticket that deals with improvements to the current exception handling so that dcac-47 can be completed 👍